src/plugins/kibana_react/public/validated_range/validated_dual_range.tsx

```diff
@@ -18,17 +18,38 @@
  */
 
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import PropTypes, { ReactNodeLike, Requireable } from 'prop-types';
+import { EuiFormRow, EuiDualRange, EuiRangeProps } from '@elastic/eui';
+import { EuiFormRowDisplayKeys } from '@elastic/eui/src/components/form/form_row/form_row';
 import { isRangeValid } from './is_range_valid';
 
-import { EuiFormRow, EuiDualRange } from '@elastic/eui';
-
 // Wrapper around EuiDualRange that ensures onChange callback is only called when range value
 // is valid and within min/max
-export class ValidatedDualRange extends Component {
-  state = {};
 
-  static getDerivedStateFromProps(nextProps, prevState) {
+interface ValidatedRangeValues extends Omit<EuiRangeProps, 'value'> {
+  value?: [number | string, number | string] | undefined;
+  allowEmptyRange?: boolean;
+  label?: string;
+  formRowDisplay?: EuiFormRowDisplayKeys;
+}
+
+interface ValidateDualRangeState {
+  isValid?: boolean;
+  errorMessage?: string;
+  value: [number | string, number | string];
+}
+
+export class ValidatedDualRange extends Component<ValidatedRangeValues> {
+  static defaultProps: { fullWidth: boolean; allowEmptyRange: boolean; compressed: boolean };
+  static propTypes: {
```

Comment:
`propTypes` should no longer be needed now that the component has been declared with a generic definition defining Props type. You can also remove `ValidatedDualRange.propTypes` on line 130. Its no longer needed now that is a typescript module.